lib/compat/wordpress-6.2/class-gutenberg-rest-pattern-directory-controller-6-2.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * REST API: Gutenberg_REST_Pattern_Directory_Controller_6_2 class
+ *
+ * @package    Gutenberg
+ * @subpackage REST_API
+ */
+
+/**
+ * Controller which provides REST endpoint for block patterns from wordpress.org/patterns.
+ */
+class Gutenberg_REST_Pattern_Directory_Controller_6_2 extends Gutenberg_REST_Pattern_Directory_Controller_6_0 {
+	/**
+	 * Search and retrieve block patterns metadata
+	 *
+	 * @since 5.8.0
+	 * @since 6.0.0 Added 'slug' to request.
+	 * @since 6.2.0 Added 'per_page', 'page', 'offset', 'order', and 'orderby' to request.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
```

Comment:
Since we override `get_items` here, we can remove it from `Gutenberg_REST_Pattern_Directory_Controller_6_0`.

Reply:
Really? I'd assumed we shouldn't to touch the older controller (except, in this case, to add the version number), to avoid confusion when porting to core.

Reply:
The code in the older controller is already in core. What we need to do is get the code from `get_items` in core and apply any new changes we want. Then when we backport for 6.2, we should just change the updated parts. Currently the `get_items` in GB_6_0 is not being called.

Reply:
So this class should be extending `WP_REST_Pattern_Directory_Controller`, not `Gutenberg_REST_Pattern_Directory_Controller_6_0`?

But I still don't understand why we would remove `get_items` from `Gutenberg_REST_Pattern_Directory_Controller_6_0` when it's not being used anymore. Wouldn't that just create extra changeset noise?

Reply:
> So this class should be extending WP_REST_Pattern_Directory_Controller, not Gutenberg_REST_Pattern_Directory_Controller_6_0?

No, it should be still extending `6_0`, since we still have some functions there(`get_transient_key`). If we didn't have any code left in 6_0, we would remove the class altogether and we would extend `WP_REST_Pattern_Directory_Controller`. 

>Wouldn't that just create extra changeset noise?

Yes, but that's okay. Compat file handling is hard for many and having an extra function there could create more confusion if left there - I've seen it in the past 😄